fuzz/fuzz_types.ml

```diff
@@ -0,0 +1,440 @@
+open Crowbar
+module T = Irmin.Type
+
+let ( |+ ) = T.( |+ )
+
+let ( |~ ) = T.( |~ )
+
+let fmt = Format.sprintf
+
+(* This is part of the standard library as of 4.08 (see [Option.map]). *)
+let option_map f o = match o with None -> None | Some t -> Some (f t)
+
+(* This is part of the standard library as of 4.08 (see [Result.map] and [Result.map_error]). *)
+let result_map fa fe r =
+  match r with Ok a -> Ok (fa a) | Error e -> Error (fe e)
+
+(* This has to be upstreamed to Crowbar. In the meantime, you can run:
+   {v opam pin crowbar https://github.com/stedolan/crowbar.git#master v}. *)
+let char = map [ bytes_fixed 1 ] (fun s -> s.[0])
+
+let string = bytes
+
+let bytes = map [ string ] Bytes.of_string
+
+let triple a b c = map [ a; b; c ] (fun a b c -> (a, b, c))
+
+(* Untyped internal representation of Irmin values. *)
+type any_value =
+  | VUnit of unit
+  | VBool of bool
+  | VChar of char
+  | VInt of int
+  | VInt32 of int32
+  | VInt64 of int64
+  | VFloat of float
+  | VString of string
+  | VBytes of bytes
+  | VList of any_value list
+  | VArray of any_value array
+  | VOption of any_value option
+  | VPair of any_value * any_value
+  | VTriple of any_value * any_value * any_value
+  | VResult of (any_value, any_value) result
+  | VRecord of dyn_record
+  | VVariant of dyn_variant
+  | VEnum of dyn_variant
+
+(* Since we can't use "real" records and variants to test how Irmin serializes
+   and deserializes them, we instead represent records with a hashtable of all
+   their fields, and variants as the name of the current constructor and its value. *)
```

Comment:
You forgot to format comments everywhere.

Reply:
@icristescu I did run `dune build @fmt --auto-promote` before committing though, that's weird. Maybe `ocamlformat` doesn't parse comments which aren't doc-comments?

Reply:
Yeah, I think not. I always check the format of these comments by hand...

Reply:
c.f. https://github.com/ocaml-ppx/ocamlformat/issues/1275

It often suffices to just make something a doc-comment. With an `.mli`, a doc comment here wouldn't show up in the documentation.